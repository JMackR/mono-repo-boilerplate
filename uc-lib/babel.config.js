module.exports = {
  presets: [["@babel/preset-env", { targets: { node: "current" } }], "module:metro-react-native-babel-preset"],
  plugins: ["@babel/plugin-proposal-class-properties", "inline-json-import"],
  sourceType: "unambiguous",
}
