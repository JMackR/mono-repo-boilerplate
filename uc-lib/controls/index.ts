export * from "./image"
export * from "./button"
export * from "./text"
export * from "./activity-indicator"
export * from "./separator"
export * from "./list"
export * from "./section-list"
export * from "./badge"
export * from "./toggle"
export * from "./login-button"
export * from "./layout"
export * from "./background"
export * from "./grid"
export * from "./touchable"
export * from "./view-block"
export * from "./text-entry"
export * from "./scroll-view"
export * from "./circle"
export * from "./slider"
export * from "./clickable"
export * from "./map"
export * from "./visibility-check"
export * from "./strike-through-text"
// export * from "./stripe-text-entry"
export * from "./input"
