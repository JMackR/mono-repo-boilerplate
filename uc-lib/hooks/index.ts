export * from "./use-infinite-scroll"
export * from "./selectable"
export * from "./screen-provider"
export * from "./loading"
