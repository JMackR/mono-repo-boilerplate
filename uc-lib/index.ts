export * from "./animations"
export * from "./assets"
export * from "./controls"
export * from "./themes"
export * from "./widgets"
export * from "./utilities"
export * from "./hooks"
export * from "./validators"
