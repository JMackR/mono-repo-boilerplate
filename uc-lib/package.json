{
  "name": "redibs-ucl",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "test": "jest --projects test/react test/react-native test/non-react"
  },
  "dependencies": {
    "@types/ramda": "^0.27.14",
    "classnames": "2.2.6",
    "google-map-react": "1.1.7",
    "hoist-non-react-statics": "3.3.2",
    "imask": "6.0.5",
    "react-compound-slider": "2.5.0",
    "react-native-autolink": "3.0.0",
    "react-native-fs": "^2.16.6",
    "zxcvbn": "4.4.2"
  },
  "devDependencies": {},
  "prettier": {
    "printWidth": 125,
    "semi": false,
    "singleQuote": false,
    "trailingComma": "all"
  },
  "jest-junit": {
    "outputDirectory": "../build",
    "ancestorSeparator": " â€º ",
    "classNameTemplate": "{classname}",
    "suiteName": "redibs-redibs-ucl jest tests",
    "suiteNameTemplate": "{filename}",
    "titleTemplate": "{title}",
    "uniqueOutputName": "true"
  }
}
