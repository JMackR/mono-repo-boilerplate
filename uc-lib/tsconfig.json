{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "baseUrl": ".",
    "outDir": "lib",
    "rootDir": "../",
    "jsx": "react-native",
    "composite": true,
    "declarationMap": true,
    "isolatedModules": true,
    "sourceMap": true,
    "moduleResolution": "Node",
    "esModuleInterop": true,
    "typeRoots": ["../node_modules/@types", "./types"]
  },
  "include": ["./"],
  "exclude": ["node_modules", "babel.config.js", "metro.config.js", "jest.config.js", "types"]
}
