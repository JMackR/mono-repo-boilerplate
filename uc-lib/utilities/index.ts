export * from "./dimensions-utility"
export * from "./media-query"
