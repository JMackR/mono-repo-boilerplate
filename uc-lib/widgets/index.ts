export * from "./account"
export * from "./attribute-list"
export * from "./breadcrumb"
export * from "./dialog/"
export * from "./empty-state"
export * from "./expanding-text-section"
export * from "./flexible-row"
export * from "./full-screen-spinner"
export * from "./inbox-row"
export * from "./indicator-dots"
export * from "./item-preview"
export * from "./listing-photo-list"
export * from "./pill-list"
export * from "./segmented-control"
export * from "./selectable-row"
export * from "./tooltip"
export * from "./user-profile"
export * from "./search"
