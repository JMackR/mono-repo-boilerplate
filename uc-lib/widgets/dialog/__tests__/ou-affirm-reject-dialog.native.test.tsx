import React from "react"
import { AffirmRejectDialog } from "uc-lib"
import { render } from "react-native-testing-library"

describe("AffirmRejectDialog tests", () => {
  let onAffirm: () => void
  let onReject: () => void
  let dismiss: () => void

  const affirmText = "Yes"
  const rejectText = "No"
  beforeEach(() => {
    onAffirm = jest.fn()
    onReject = jest.fn()
    dismiss = jest.fn()
  })

  test("Render content", () => {
    const tree = render(
      <AffirmRejectDialog
        onAffirm={onAffirm}
        onReject={onReject}
        affirmText={affirmText}
        rejectText={rejectText}
        dismiss={dismiss}
        title="Test Title"
        dismissOnReject={true}
        body="Test Body"
      />,
    )
    expect(tree).toMatchSnapshot()
  })
  test("Affirm button tapped", () => {
    const tree = render(
      <AffirmRejectDialog
        onAffirm={onAffirm}
        onReject={onReject}
        affirmText={affirmText}
        rejectText={rejectText}
        dismiss={dismiss}
        title="Test Title"
        dismissOnReject={true}
        body="Test Body"
      />,
    )
    const affirmButton = tree.getByTestId("uc-lib.affirm-reject-dialog.affirm-button")
    affirmButton.props.onClick()
    expect(onAffirm).toBeCalledTimes(1)
    expect(dismiss).toBeCalledTimes(1)
    expect(onReject).toHaveBeenCalledTimes(0)
  })
  test("Reject button tapped", () => {
    const tree = render(
      <AffirmRejectDialog
        onAffirm={onAffirm}
        onReject={onReject}
        affirmText={affirmText}
        rejectText={rejectText}
        dismiss={dismiss}
        title="Test Title"
        dismissOnReject={true}
        body="Test Body"
      />,
    )
    const rejectButton = tree.getByTestId("uc-lib.affirm-reject-dialog.reject-button")
    rejectButton.props.onClick()
    expect(onReject).toBeCalledTimes(1)
    expect(dismiss).toBeCalledTimes(1)
    expect(onAffirm).toHaveBeenCalledTimes(0)
  })
})
