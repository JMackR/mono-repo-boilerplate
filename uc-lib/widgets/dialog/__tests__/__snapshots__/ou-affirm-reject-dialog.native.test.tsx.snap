// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AffirmRejectDialog tests Render content 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "backgroundColor": "rgba(18,18,18,0.3)",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": undefined,
        "flexBasis": "auto",
        "flexDirection": "row",
        "flexGrow": 0,
        "flexShrink": 1,
        "height": undefined,
        "justifyContent": "flex-start",
        "margin": 16,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "width": undefined,
      }
    }
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 8,
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "stretch",
            "backgroundColor": undefined,
            "flexBasis": "auto",
            "flexDirection": "row",
            "flexGrow": 0,
            "flexShrink": 1,
            "height": undefined,
            "justifyContent": "flex-start",
            "margin": 16,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "width": undefined,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "stretch",
              "backgroundColor": undefined,
              "flexBasis": "auto",
              "flexDirection": "column",
              "flexGrow": 1,
              "flexShrink": 1,
              "height": undefined,
              "justifyContent": "flex-start",
              "overflow": undefined,
              "width": undefined,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "stretch",
                "backgroundColor": undefined,
                "flexBasis": "auto",
                "flexDirection": "row",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": undefined,
                "justifyContent": "flex-start",
                "margin": undefined,
                "marginBottom": undefined,
                "marginLeft": 16,
                "marginRight": 16,
                "marginTop": undefined,
                "width": undefined,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": undefined,
                  "flexBasis": "auto",
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "flexWrap": undefined,
                  "height": undefined,
                  "justifyContent": "center",
                  "overflow": undefined,
                  "width": undefined,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "stretch",
                    "backgroundColor": undefined,
                    "flexBasis": "auto",
                    "flexDirection": "column",
                    "flexGrow": 0,
                    "flexShrink": 1,
                    "height": undefined,
                    "justifyContent": "flex-start",
                    "overflow": undefined,
                    "width": undefined,
                  }
                }
              >
                <Text
                  accessibilityLabel="redibs-ucl.affirm-reject-dialog.title"
                  style={
                    Array [
                      Object {
                        "fontFamily": "Mulish-Bold",
                        "fontSize": 20,
                        "fontWeight": "900",
                        "lineHeight": 24,
                      },
                      Object {
                        "color": "#121212",
                        "textAlign": "center",
                        "textDecorationLine": undefined,
                      },
                    ]
                  }
                  testID="redibs-ucl.affirm-reject-dialog.title"
                >
                  Test Title
                </Text>
                <View
                  style={
                    Object {
                      "height": 16,
                      "width": undefined,
                    }
                  }
                />
                <Text
                  accessibilityLabel="redibs-ucl.affirm-reject-dialog.body"
                  style={
                    Array [
                      Object {
                        "fontFamily": "Mulish-Regular",
                        "fontSize": 16,
                        "fontWeight": "normal",
                        "lineHeight": 20,
                      },
                      Object {
                        "color": "#121212",
                        "textAlign": "center",
                        "textDecorationLine": undefined,
                      },
                    ]
                  }
                  testID="redibs-ucl.affirm-reject-dialog.body"
                >
                  Test Body
                </Text>
                <View
                  style={
                    Object {
                      "height": 16,
                      "width": undefined,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "height": 16,
                      "width": undefined,
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "stretch",
                "backgroundColor": undefined,
                "flexBasis": "auto",
                "flexDirection": "row",
                "flexGrow": 0,
                "flexShrink": 1,
                "height": undefined,
                "justifyContent": "flex-start",
                "overflow": undefined,
                "width": undefined,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "stretch",
                  "backgroundColor": undefined,
                  "flexBasis": "auto",
                  "flexDirection": "column",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "flexWrap": undefined,
                  "height": undefined,
                  "justifyContent": "flex-start",
                  "overflow": undefined,
                  "width": undefined,
                }
              }
            >
              <View
                accessibilityLabel="redibs-ucl.affirm-reject-dialog.reject-button"
                accessibilityRole="button"
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignContent": "stretch",
                    "alignItems": "center",
                    "backgroundColor": "rgba(226, 226, 226, 1)",
                    "borderRadius": 4,
                    "flex": 0,
                    "flexDirection": "row",
                    "flexWrap": "nowrap",
                    "height": 40,
                    "justifyContent": "center",
                    "paddingHorizontal": 8,
                  }
                }
                testID="redibs-ucl.affirm-reject-dialog.reject-button"
              >
                <View
                  style={
                    Object {
                      "alignSelf": "auto",
                      "flex": 0,
                      "flexDirection": "column",
                    }
                  }
                >
                  <Text
                    accessibilityLabel="redibs-ucl.button.title"
                    style={
                      Array [
                        Object {
                          "alignContent": "center",
                          "textAlign": "center",
                        },
                        Object {
                          "fontFamily": "Mulish-Bold",
                          "fontSize": 16,
                          "fontWeight": "bold",
                          "lineHeight": 20,
                        },
                        Object {
                          "color": "#121212",
                        },
                      ]
                    }
                    testID="redibs-ucl.button.title"
                  >
                    No
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "height": undefined,
                  "width": 8,
                }
              }
            />
            <View
              style={
                Object {
                  "alignItems": "stretch",
                  "backgroundColor": undefined,
                  "flexBasis": "auto",
                  "flexDirection": "column",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "flexWrap": undefined,
                  "height": undefined,
                  "justifyContent": "flex-start",
                  "overflow": undefined,
                  "width": undefined,
                }
              }
            >
              <View
                accessibilityLabel="redibs-ucl.affirm-reject-dialog.affirm-button"
                accessibilityRole="button"
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignContent": "stretch",
                    "alignItems": "center",
                    "backgroundColor": "rgba(0, 168, 126, 1)",
                    "borderRadius": 4,
                    "flex": 0,
                    "flexDirection": "row",
                    "flexWrap": "nowrap",
                    "height": 40,
                    "justifyContent": "center",
                    "paddingHorizontal": 8,
                  }
                }
                testID="redibs-ucl.affirm-reject-dialog.affirm-button"
              >
                <View
                  style={
                    Object {
                      "alignSelf": "auto",
                      "flex": 0,
                      "flexDirection": "column",
                    }
                  }
                >
                  <Text
                    accessibilityLabel="redibs-ucl.button.title"
                    style={
                      Array [
                        Object {
                          "alignContent": "center",
                          "textAlign": "center",
                        },
                        Object {
                          "fontFamily": "Mulish-Bold",
                          "fontSize": 16,
                          "fontWeight": "bold",
                          "lineHeight": 20,
                        },
                        Object {
                          "color": "#FFFFFF",
                        },
                      ]
                    }
                    testID="redibs-ucl.button.title"
                  >
                    Yes
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
